lib/admin/parsers/workbook.rb

```diff
@@ -13,31 +17,49 @@ def initialize(rubyxl_workbook,
       end
 
       def call
-        as_hashes = transform_worksheet
+        [ parse_mediators, processed_blacklist ]
+      end
+
+      private
+
+      def parse_mediators
+        as_hashes = transform_mediators
         as_hashes = @data_parser.parse(as_hashes) if @data_parser
         as_hashes
       end
 
-      private
+      def parse_blacklist
+        blacklist_worksheet[1..-1].map { |row| row.cells.first.value }
+      end
 
-      def transform_worksheet
-        return [] if worksheet[0].nil?
+      def processed_blacklist
+        @headings_processor.process(parse_blacklist)
+      end
 
-        processed_headings = @headings_processor.process(
-          worksheet[0].cells.map { |cell| cell.value }
-        )
+      def transform_mediators
+        return [] if mediators_worksheet[0].nil?
 
-        worksheet[1..-1].map do |row|
+        mediators_worksheet[1..-1].map do |row|
           row.cells.each_with_index.inject({}) do |hash, (cell, index)|
             hash.merge({processed_headings[index].to_sym => cell.value.to_s})
           end
         end
       end
 
-      def worksheet
-        @rubyxl_workbook[WORKSHEET] if @rubyxl_workbook
+      def processed_headings
+        @headings_processor.process(
+          mediators_worksheet[0].cells.map { |cell| cell.value }
+        )
+      end
+
+      def mediators_worksheet
```

Comment:
Would an OO approach be to create an object to represent the worksheets, encapsulating the different behaviour in the 'mediator' workbook and the 'metadata' workbook?